client/tests/integration/components/packages/attachments-test.js

```diff
@@ -0,0 +1,129 @@
+          id: 'f0f2f3a3-3936-499b-8f37-a9827a1c14f2',
+        },
+      ],
+    };
+
+    await render(hbs`<
+      Packages::Attachments
+      @package={{this.package}}
+     />`);
+
+    await assert.equal(findAll('[data-test-document-to-be-deleted-name]').length, 0);
+
+    // mark a file for deletion
+    await click('[data-test-delete-file-button="0"]');
+
+    await assert.equal(findAll('[data-test-document-name]').length, 1);
+    await assert.equal(findAll('[data-test-document-to-be-deleted-name]').length, 1);
+    await assert.equal(find('[data-test-document-to-be-deleted-name]').textContent.trim(), 'PAS Form.pdf');
+
+    // mark second file for deletion
+    await click('[data-test-delete-file-button="0"]');
+
+    await assert.equal(findAll('[data-test-document-name]').length, 0);
+    await assert.equal(findAll('[data-test-document-to-be-deleted-name]').length, 2);
+    await assert.equal(find('[data-test-document-to-be-deleted-name="0"]').textContent.trim(), 'PAS Form.pdf');
+    await assert.equal(find('[data-test-document-to-be-deleted-name="1"]').textContent.trim(), 'Action Changes.excel');
+
+    // unmark a file for deletion
+    await click('[data-test-undo-delete-file-button="1"]');
+
+    await assert.equal(findAll('[data-test-document-name]').length, 1);
+    await assert.equal(findAll('[data-test-document-to-be-deleted-name]').length, 1);
+    await assert.equal(find('[data-test-document-name]').textContent.trim(), 'Action Changes.excel');
+
+    // unmark another file for deletion
+    await click('[data-test-undo-delete-file-button="0"]');
+
+    await assert.equal(findAll('[data-test-document-name]').length, 2);
+    await assert.equal(findAll('[data-test-document-to-be-deleted-name]').length, 0);
+    await assert.equal(find('[data-test-document-name="0"]').textContent.trim(), 'Action Changes.excel');
+    await assert.equal(find('[data-test-document-name="1"]').textContent.trim(), 'PAS Form.pdf');
+  });
+
+  test('user can select local files for upload', async function(assert) {
+    this.package = {
+      id: '123',
+      documents: [
+      ],
+    };
+
+    await render(hbs`<
+      Packages::Attachments
+      @package={{this.package}}
+     />`);
+
+
+    const file = new File(['foo'], 'PAS Form.pdf', { type: 'text/plain' });
+    const file2 = new File(['foo'], 'Action Changes.excel', { type: 'text/plain' });
+
+    await render(hbs`<
+      Packages::Attachments
+      @package={{this.package}}
+    />`);
+
+    await assert.equal(findAll('[data-test-document-to-be-uploaded-name]').length, 0);
+
+    await selectFiles('#FileUploader123 > input', file, file2);
+
+    await assert.equal(findAll('[data-test-document-to-be-uploaded-name]').length, 2);
+    await assert.equal(find('[data-test-document-to-be-uploaded-name="0"]').textContent.trim(), 'PAS Form.pdf');
+    await assert.equal(find('[data-test-document-to-be-uploaded-name="1"]').textContent.trim(), 'Action Changes.excel');
```

Comment:
Check out some of these nicer DOM assertion APIs: https://github.com/simplabs/qunit-dom